### PR TITLE
Add some docs explaining mongo db and collection auto-creation

### LINKED ISSUE #324: Mongo db name incorrect
Things are going into a db named `db` rather than what is in the config.

### PR DESCRIPTION
**Pull request by wtgee**
Closes #324

### CODE REVIEW

File: pocs/utils/database.py [hunk 1]
```diff
@@ -57,9 +69,10 @@ def __init__(self, db='panoptes', host='localhost', port=27017, connect=False):
             'weather',
         ]
 
-        db_handle = getattr(self._client, db)
+        # Create an attribute on the client with the db name.
+        db_handle = self._client[db]
 
-        # Setup static connections to the collections we want
+        # Setup static connections to the collections we want.
```
Comment (jamessynge) [inline pocs/utils/database.py]: Does the `getattr(db_handle, collection)` create the collection? Or is that lazy too (i.e. maybe pymongo doesn't do anything until we try to read or write from the collection)?

No need to change this, but if the operation isn't lazy, and we wanted it to be, we could instead add a `__getattr__` method that validates that the attr is a member of `self.collections` before doing the setattr.

File: pocs/utils/database.py [hunk 2]
```diff
@@ -46,6 +57,7 @@ def __init__(self, db='panoptes', host='localhost', port=27017, connect=False):
         # Get the mongo client
         self._client = get_shared_mongo_client(host, port, connect)
 
+        # Pre-defined list of collections that are valid.
```
Comment (jamessynge) [inline pocs/utils/database.py]: Is self.collections intended for use by the caller?

File: pocs/utils/database.py [hunk 3]
```diff
@@ -36,6 +36,17 @@ def __init__(self, db='panoptes', host='localhost', port=27017, connect=False):
         starts and can be read and updated as the project is running. The server
         is a wrapper around a mongodb collection.
 
+        Note:
```
Comment (jamessynge) [inline pocs/utils/database.py]: Thank you!

File: pocs/utils/database.py (hunk 2 above)
Comment (wtgee) [reply pocs/utils/database.py]: I didn't think it needed to be private in case, but I suppose it could be.

File: pocs/utils/database.py (hunk 1 above)
Comment (wtgee) [reply pocs/utils/database.py]: It's lazy and happens upon first read/write.

Comment (wtgee) [reply pocs/utils/database.py]: I'm just going to leave as is for now. Probably a few more things we could do to improve overall but nothing necessary right now.

**Pull request merged**